hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/bucket/OMBucketDeleteRequest.java

```diff
@@ -69,15 +65,21 @@ public OMClientResponse validateAndUpdateCache(OzoneManager ozoneManager,
     String volumeName = omRequest.getDeleteBucketRequest().getVolumeName();
     String bucketName = omRequest.getDeleteBucketRequest().getBucketName();
 
-    // acquire lock
-    omMetadataManager.getLock().acquireBucketLock(volumeName, bucketName);
-
     // Generate end user response
     OMResponse.Builder omResponse = OMResponse.newBuilder()
         .setStatus(OzoneManagerProtocolProtos.Status.OK)
         .setCmdType(omRequest.getCmdType());
 
     try {
+      // check Acl
+      if (ozoneManager.getAclsEnabled()) {
+        checkAcls(ozoneManager, OzoneObj.ResourceType.BUCKET,
```

Comment:
same as above. we also need to acquire read lock on the  bucket when checking acl.

Reply:
Right now lock manager does not have readlock acquire on bucket. Do you mean acquire bucket lock?

And why do we need to acquire lock here? I don't see for any requests acquire of lock in current OzoneManager.java code.

Reply:
Generally, we don't need bucket lock to check bucket acls against the callers. However, putting a bucket lock for checkacls prevent bucket acls (and bucket) from being modified when checking.

Reply:
For now, will add a TODO, once read locks are supported will get readLock and then call checkAcls.